packages/flutter/lib/src/material/outlined_button_theme.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'button_style.dart';
+import 'theme.dart';
+
+/// A [ButtonStyle] that overrides the default appearance of
+/// [OutlinedButton]s when it's used with [OutlinedButtonTheme] or with the
+/// overall [Theme]'s [ThemeData.outlinedButtonTheme].
+///
+/// The [style]'s properties override [OutlinedButton]'s default style,
+/// i.e.  the [ButtonStyle] returned by [OutlinedButton.defaultStyleOf]. Only
+/// the style's non-null property values or resolved non-null
+/// [MaterialStateProperty] values are used.
+///
+/// See also:
+///
+///  * [OutlinedButtonTheme], the theme which is configured with this class.
+///  * [OutlinedButton.defaultStyleOf], which returns the default [ButtonStyle]
+///    for text buttons.
+///  * [OutlinedButton.styleOf], which converts simple values into a
+///    [ButtonStyle] that's consistent with [OutlinedButton]'s defaults.
+///  * [MaterialStateProperty.resolve], "resolve" a material state property
+///    to a simple value based on a set of [MaterialState]s.
+///  * [ThemeData.outlinedButtonTheme], which can be used to override the default
+///    [ButtonStyle] for [OutlinedButton]s below the overall [Theme].
+@immutable
+class OutlinedButtonThemeData with Diagnosticable {
```

Comment:
Is the reason for having this class not just be a `ButtonStyle` is because we would want to a place for future parameters that are not part of the style but are specific to the `OutlinedButton`?

Reply:
That's a possibility, but the real reason was just to conform to the naming conventions for component themes and their configuration data classes.